.circleci/config.yml

```diff
@@ -102,3 +215,14 @@ workflows:
       - test_frontend:
           requires:
             - build
+      - deploy:
+          requires:
+            - lint
+            - test_backend
+            - test_frontend
+          filters:
+            branches:
+              only:
+                - << pipeline.parameters.dev_git_branch >>
+                - << pipeline.parameters.staging_git_branch >>
+                - << pipeline.parameters.prod_git_branch >>
```

Comment:
FWIW I'm not completely pleased with this filtering; it's not exactly skip/fail fast. This limits the deploy job to any branch named `main` or `production`.  In the deploy job, after assets are built and cf is installed, I then match for repo name.  This isn't really a security concern, as only our repo has access to dev deployment variables and only your repo has access to staging/prod vars.